src/dmd/expression.d

```diff
@@ -5454,7 +5462,27 @@ extern (C++) final class VectorExp : UnaExp
         v.visit(this);
     }
 }
+extern (C++) final class WithExp : UnaExp
```

Comment:
Implement a `inout(WithExp) isWithExp() { return op == EXP.withExp ? cast(typeof(return))this : null; }`